.github/workflows/build.yml

```diff
@@ -64,8 +56,108 @@ jobs:
 
       - shell: bash
         id: test
-        continue-on-error: ${{ matrix.continue-on-error }}
-        name: Test
+        name: Integration Tests
+        run: |
+          mkdir -p ~/.stack
+          echo "system-ghc: true" >> ~/.stack/config.yaml
+          .github/ci.sh test_dist
+      
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: "saw-${{ runner.os }}-${{ matrix.ghc }}"
+          path: dist/bin/saw
+
+      - shell: bash
+        if: "steps.test.outcome == 'failure'"
+        name: Warn if tests failed
+        run: echo "::error ::Test Suite Failed. Pipeline allowed to pass until tests are reliable."
+
+  s2n-tests:
+    name: "Test s2n proofs"
+    needs: build
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        s2n-target:
+          - hmac
+          - drbg
+          - sike
+          - bike
+          - tls
+          - hmac-failure
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          mkdir -p s2nTests/bin
+
+      - name: Download previously-built SAW 
+        uses: actions/download-artifact@v2
+        with:
+          name: "saw-Linux-8.6.5"
```

Comment:
Are we supposed to run the s2n tests only for a single s2n? Or should they be ran for every s2n we generate? I was expecting to see the same matrix specification of GHC versions that we have for the build. Version differences between GHC can sometimes result in correctness differences, and they can certainly result in performance differences,so it'd be nice to catch those if possible.